apps/keyboard/js/keyboard/layout_rendering_manager.js

```diff
@@ -127,14 +127,24 @@ LayoutRenderingManager.prototype.updateLayoutRendering = function() {
     ((typeof currentIMEngine.displaysCandidates !== 'function') ||
       currentIMEngine.displaysCandidates()));
 
-  var p = new Promise(function(resolve) {
-    this.app.viewManager.render(currentPage, {
-      uppercase: this.app.upperCaseStateManager.isUpperCase,
-      inputType: this.app.getBasicInputType(),
-      showCandidatePanel: needsCandidatePanel
-    }, resolve);
-  }.bind(this)).then(this._afterRenderDrew.bind(this));
+  var p = Promise.resolve(currentPage);
 
+  var interpolaters = currentIMEngine.pageInterpolaters;
+  if (currentPage.interpolater && interpolaters &&
+    typeof interpolaters[currentPage.interpolater] === 'function') {
+      p.then(interpolaters[currentPage.interpolater]);
+  }
+
+  p.then((currentPage) => {
+    return new Promise((resolve) => {
+      this.app.viewManager.render(currentPage, {
```

Comment:
Please ensure `this._updateHeight()` at line 152 always happen right after this call. Since you moved this inside a `then()`, `this._updateHeight()` will now happen before this.